### Assignment- Week 5

#### What is the northernmost airport in the United States?


In [6]:
import pandas as pd
northern_airport = pd.read_csv('https://raw.githubusercontent.com/maddogdatascience/DATA607/master/airports.csv')
northern_airport.head()
northern_airport.sort_values("lat", ascending = False).head(5)

faa                                        name        lat         lon  \
393  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
217  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
98   AIN                          Wainwright Airport  70.638056 -159.994722   
674  K03                               Wainwright As  70.613378 -159.860350   
140  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst  
393  149  -5   A  
217   44  -9   A  
98    41  -9   A  
674   35  -9   A  
140   96  -9   A

> Based google maps lat of Dillant Hopkins Airport is 42.89 and Longtitue is -72.27. The data provided seems wrong so **"Wiley Post Will ROgers Mem"** is the northemost airport in the United Stats. Source: https://www.google.com/maps/place/Dillant-Hopkins+Airport-Een/@42.9062734,-72.2726591,21z/data=!4m8!1m2!2m1!1sdillant+hopkins+airport!3m4!1s0x0:0x7d401c5a9d69b70f!8m2!3d42.9064479!4d-72.2726201

#### What is the easternmost airport in the United States?


In [7]:
eastern = pd.read_csv('https://raw.githubusercontent.com/maddogdatascience/DATA607/master/airports.csv')
eastern_airport = eastern[(eastern['lat'] > 44) & (eastern['lat'] <= 50) & (eastern['lon'] >= -75) & (eastern['lon'] <= -65)]
eastern_airport.head()
eastern_airport.sort_values('lon', ascending=False).head(5)

faa                                 name        lat        lon  alt  tz  \
419   EPM           Eastport Municipal Airport  44.910111 -67.012694   45  -5   
591   HUL                         Houlton Intl  46.123083 -67.792056  489  -4   
246   CAR                         Caribou Muni  46.871500 -68.017917  626  -4   
1052  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797  534  -4   
1340  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778  988  -5   

     dst  
419    A  
591    A  
246    A  
1052   A  
1340   A

> Since the latitude and longitudinal values seemed accurate, I double-checked each value using www.airnav.com and www.latitude.to Each value matched the values in the table above. **Eastport Municipal Airport** is the earthernmost airport in the United States.

#### On February 12th, 2013, which New York area airport had the windiest weather?


In [8]:
windy = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')
# sorting data based on requirements
windy_weather = windy[(windy["month"]==2) & (windy["day"]==12) & (windy["year"]==2013)]
windy_weather.head()
windy_weather.sort_values("wind_speed", ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

> Since, LGA and JFK is the airport of NYC, **LGA** had the most windiest weather on Feb 12th , 2013.